mm2src/mm2_main/src/lp_network.rs

```diff
@@ -138,36 +139,94 @@ async fn process_p2p_message(
     mut message: GossipsubMessage,
     i_am_relay: bool,
 ) {
+    fn is_valid(topics: &[TopicHash]) -> Result<(), String> {
+        if topics.is_empty() {
+            return Err("At least one topic must be provided.".to_string());
+        }
+
+        let first_topic_prefix = topics[0].as_str().split(TOPIC_SEPARATOR).next().unwrap_or_default();
+        for item in topics.iter().skip(1) {
+            if !item.as_str().starts_with(first_topic_prefix) {
+                return Err(format!(
+                    "Topics are invalid, received more than one topic kind. Topics '{:?}",
+                    topics
+                ));
+            }
+        }
+
+        Ok(())
+    }
+
     let mut to_propagate = false;
-    let mut orderbook_pairs = vec![];
 
     message.topics.dedup();
     drop_mutability!(message);
 
-    for topic in message.topics {
+    if let Err(err) = is_valid(&message.topics) {
+        log::error!("{}", err);
+        return;
+    }
+
+    let inform_about_break = |used: &str, all: &[TopicHash]| {
+        log::debug!(
+            "Topic '{}' proceed and loop is killed. Whole topic list was '{:?}'",
+            used,
+            all
+        );
+    };
```

Comment:
Why not debug the number of topics instead of the topic list, this can be a very long message if we debug the whole list.